packages/plugin-ext/src/plugin/plugin-context.ts

```diff
@@ -278,6 +278,21 @@ import { NotebookEditorsExtImpl } from './notebook/notebook-editors';
 import { TestingExtImpl } from './tests';
 import { UriExtImpl } from './uri-ext';
 
+export function createAPIObject<T extends Object>(rawObject: T): T {
+    return new Proxy({}, {
+        get(target, p, receiver) {
+            const isOwnProperty = !!Object.getOwnPropertyDescriptor(rawObject, p);
+            const val = Reflect.get(rawObject, p);
+            if (!isOwnProperty && typeof val === 'function') {
+                // bind functions that are inherited from the prototype to the object itself.
+                // This should handle the case of events.
+                return val.bind(rawObject);
+            }
+            return val;
+        },
+    }) as T;
```

Comment:
Suggestion: Using the `rawObject` as the target of the proxy seems to fix the tests. I assume some code (either our own or some plugin code) relies on more than just the `get` functionality of the proxy. Stuff like `has` (used for the `in` operator) should be implemented as well. 
```suggestion
    return new Proxy(rawObject, {
        get(target, p) {
            const isOwnProperty = !!Object.getOwnPropertyDescriptor(target, p);
            const val = Reflect.get(target, p);
            if (!isOwnProperty && typeof val === 'function') {
                // bind functions that are inherited from the prototype to the object itself.
                // This should handle the case of events.
                return val.bind(target);
            }
            return val;
        },
    }) as T;
```

Reply:
The idea was to make sure users of the proxy did not have access to the wrapped object. I believe it's impossible to access the target object when you have a proxy object, but I'm not 100% sure. Do you know more about this topic?

Reply:
As far as I'm aware, the JS API does not expose the target of a `Proxy` object to outsiders.